tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/namespaces/NamespaceController.java

```diff
@@ -121,4 +121,27 @@ private ModelAndView getRemoveNamespaceResult(HttpServletRequest request, String
 
 		return new ModelAndView(EmptySuccessView.getInstance());
 	}
+
+	private void validateUpdateNamespaceData(String prefix, String namespace) throws ClientHTTPException {
+		if (namespace.length() == 0) {
+			throw new ClientHTTPException(SC_BAD_REQUEST, "No namespace name found in request body");
+		}
+
+		if (!isEmpty(prefix) && !isAlphanumeric(prefix)) {
+			throw new ClientHTTPException(SC_BAD_REQUEST, "Prefix not alphanumeric");
+		}
+
+		if (!isValidNamespaceIri(namespace)) {
+			throw new ClientHTTPException(SC_BAD_REQUEST, "Namespace not valid");
+		}
+	}
+
+	private boolean isValidNamespaceIri(String namespace) {
+		try {
+			return ParsedIRI.create(namespace).isAbsolute();
```

Comment:
Looks good. 

The only thing I'm not sure about is if you should be using `ParsedIRI.create` here instead of the constructor (`new ParsedIRI`). The difference between the two is that `create` attempts to percent-encode any unexpected characters it encounters in the input (only throwing a parse error if even after doing this the IRI is still illegal), whereas the constructor just throws an error immediately if the input is not valid.

So for example if your input is `http://example.org/with whitespace`, the constructor would throw an error (whitespaces are not legal in IRIs), but the `#create` method would return `http://example.org/with%20whitespace`.

Which is the right choice here kind of depends on how strict you want to be.

Reply:
Thanks for the explanation! Didn't know about the difference, I think it would be better to have a stricter validation, just changed my code to the constructor variant and added your two examples to my tests.